SpiffyTitles/plugin.py

```diff
@@ -430,17 +430,43 @@ def t(self, irc, msg, args, query):
         url = self.get_url_from_message(message)
         title = None
         error_message = self.registryValue("onDemandTitleError", channel=channel)
+        handled = False
+
+        if url:
+            title = self.get_title_by_url(query, channel)
+            if title is not None and title:
+                if "#" in channel:
+                    """
+                    This prevents the title being sent twice, when t
+                    is used in a channel, if it is handled by enabled
+                    handlers already, t will pass silently.
+                    If t is requested in a /msg, the title acquired
+                    using the enabled handlers will be replied back.
+                    """
+                    handled = True
+            else:
+                """
+                If the handlers are disabled, use the default handler
+                and attempt to get the title anyway.
+                """
+                title = self.handler_default(url, channel, t_override=True)
 
-        try:
-            if url:
-                title = self.get_title_by_url(query, channel)
-        except:
-            pass
-
-        if title is not None and title:
-            irc.queueMsg(ircmsgs.privmsg(channel, title))
+            if not handled:
+                if title is not None and title:
+                    if "#" in channel:
```

Comment:
Limnoria has a way to check if something is a channel without doing something like this. There is something to check if it's a PM too. Check my other repos if you can't find it.